packages/language/src/linking/resolver.ts

```diff
@@ -56,6 +56,19 @@ export class ReferencesCache {
   }
 }
 
+// Returns the qualified name in reverse order, e.g. "A.B.C" -> ["C", "B", "A"]
+export function getQualifiedName(reference: Reference): string[] {
+  if (reference.owner.container?.kind === SyntaxKind.MemberCall) {
+    const memberCall = reference.owner.container;
+    if (memberCall.previous?.element?.ref) {
+      const names = getQualifiedName(memberCall.previous.element.ref);
+      return [reference.text, ...names];
```

Comment:
Suggestion: Since we don't reuse the array anywhere else, can we do this instead? This performs an in-place modification, which should be faster.
```suggestion
      names.unshift(reference.text);
      return names;
```

Reply:
That makes sense! I made a change which removed all destructuring operations.